python/ray/_private/utils.py

```diff
@@ -1899,3 +1899,20 @@ def try_import_each_module(module_names_to_import: List[str]) -> None:
             importlib.import_module(module_to_preload)
         except ImportError:
             logger.exception(f'Failed to preload the module "{module_to_preload}"')
+
+
+def update_envs(env_vars: Dict[str, str]):
+    if env_vars is None:
+        return
+    replaceable_keys = [
+        "PATH",
+        "LD_LIBRARY_PATH",
+        "DYLD_LIBRARY_PATH",
+        "LD_PRELOAD",
+    ]
+
+    for key, value in env_vars.items():
+        if key in replaceable_keys and key in os.environ and isinstance(value, str):
```

Comment:
There's a weird behavior here: If `key in replaceable_keys` and `key in os.environ` but `value` is not a str, then it will reset the env var. This would be surprising to users of this function.

Can we raise an exception in this case, or simply `str(value)`?

Reply:
I detected the context. This env_vars must be str. My check was redundant and I have removed it.